### PR TITLE
Test Case: Add check suspend resume with audio

### PR DESCRIPTION
**Pull request by Bin-QA**
Add new Test Case for the suspend resume
playback/capture on the backgroud, do the suspend resume

Signed-off-by: Wu, BinX <binx.wu@intel.com>

### CODE REVIEW

**Comment by wenqingfu** [comment]
Can it be done with by creating a "combo" test case by reusing existing ones, such as:

check-playback.sh &
check-suspend-resume.sh

with some parameters tweaking, of course.

**Comment by Bin-QA** [comment]
> Can it be done with by creating a "combo" test case by reusing existing ones, such as:
> 
> check-playback.sh &
> check-suspend-resume.sh
> 
> with some parameters tweaking, of course.

too many options,  
on the other hand, suspend and resume support random value,  
so not sure playback/capture can be auto finish after resume
@xiulipan @keqiaozhang @libinyang @aiChaoSONG can you give me some advice

**Comment by xiulipan** [comment]
@wenqingfu It would be hard to sync the playback test with the suspend resume test.
I think this would be a good solution for both developer and user.

File: test-case/check-suspend-resume-with-audio.sh [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        sleep_lst[$i]=${OPT_VALUE_lst['S']}
+        wait_lst[$i]=${OPT_VALUE_lst['w']}
+    done
+fi
+
+func_suspend_resume()
+{
+    local sleep_t=$1 wait_t=$2
+    local sleep_count wake_count
+    # cleanup dmesg befor run case
```
Comment (stripes416) [inline test-case/check-suspend-resume-with-audio.sh]: befor -> before

File: test-case/check-suspend-resume-with-audio.sh [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    dev=$(func_pipeline_parse_value $idx dev)
+    pcm=$(func_pipeline_parse_value $idx pcm)
+    type=$(func_pipeline_parse_value $idx type)
+    dlogi "Run $TYPE command for the background"
+    dlogc $CMD -D$dev -r $rate -c $channel -f $fmt $FILE -q
+    $CMD -D$dev -r $rate -c $channel -f $fmt $FILE -q &
+    for i in $(seq 1 $loop_count)
+    do
+        dlogi "Round($i/$loop_count)"
+        # cleanup dmesg befor run case
```
Comment (stripes416) [inline test-case/check-suspend-resume-with-audio.sh]: befor -> before

File: test-case/check-suspend-resume-with-audio.sh [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+dlogi "Current suspend/resume type mode: $(cat /sys/power/mem_sleep)"
+
+loop_count=${OPT_VALUE_lst['l']}
+
+if [ "${OPT_VALUE_lst['m']}" == "playback" ]; then
+    CMD='aplay'     FILE='/dev/zero'    TYPE='playback,both'
+elif [ "${OPT_VALUE_lst['m']}" == "capture" ]; then
+    CMD='arecord'   FILE='/dev/null'    TYPE='capture,both'
+else
+    dlogw "Error alas application type: ${OPT_VALUE_lst['m']}"
```
Comment (stripes416) [inline test-case/check-suspend-resume-with-audio.sh]: What does alas mean here?

**Comment by stripes416** [comment]
The main differences between this automated test & the manual version are that the manual version (for playback) is run using a raw audio file with the operator listening for problems with the audio after the suspend/resume cycle occurs. 

I'd suggest adding some checking to make sure that the aplay/arecord processes are still alive and well before looping for another suspend/resume cycle. Other than that, the test cases are fairly similar.

File: test-case/check-suspend-resume-with-audio.sh (hunk 3 above)
Comment (plbossart) [reply test-case/check-suspend-resume-with-audio.sh]: alas alsa?

**Comment by stripes416** [comment]
@Bin-QA I'm a little confused -- The comments from me & Pierre have been marked resolved, but there doesn't seem to be an update. Maybe you forgot to resubmit?

**Comment by xiulipan** [comment]
> @Bin-QA I'm a little confused -- The comments from me & Pierre have been marked resolved, but there doesn't seem to be an update. Maybe you forgot to resubmit?
@Bin-QA I also can confirm the issue, I think you forget to push the update, as there is not update on the PR after you first submit.

**Comment by Bin-QA** [comment]
> @Bin-QA I'm a little confused -- The comments from me & Pierre have been marked resolved, but there doesn't seem to be an update. Maybe you forgot to resubmit?

I already update it, but patch is not submit, I need verify the change to submit it

**Comment by Bin-QA** [comment]
@stripes416 @xiulipan already update the commit

**Comment by stripes416** [comment]
> @stripes416 @xiulipan already update the commit
Awesome! Thanks @Bin-QA

File: test-case/check-suspend-resume-with-audio.sh [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+OPT_PARM_lst['f']=1         OPT_VALUE_lst['f']=''
+
+func_opt_parse_option $*
+func_lib_check_sudo
+func_lib_setup_kernel_last_line
+
+tplg=${OPT_VALUE_lst['t']}
+[[ ${OPT_VALUE_lst['s']} -eq 1 ]] && func_lib_start_log_collect
+
+# overwrote the subscript: test-case LOG_ROOT environment
```
Comment (stripes416) [inline test-case/check-suspend-resume-with-audio.sh]: overwrite maybe?

File: test-case/check-suspend-resume-with-audio.sh (hunk 3 above)
Comment (stripes416) [reply test-case/check-suspend-resume-with-audio.sh]: I think you missed an update :)

**Comment by stripes416** [comment]
This is definitely an interesting solution to adding audio to the suspend/resume check --- no need to maintain two copies of the same basic test.

File: test-case/check-suspend-resume-with-audio.sh [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    dev=$(func_pipeline_parse_value $idx dev)
+    pcm=$(func_pipeline_parse_value $idx pcm)
+    dlogi "Run $TYPE command for the background"
+    dlogc "$cmd -D$dev -r $rate -c $channel -f $fmt $file_name -q"
+    $cmd -D$dev -r $rate -c $channel -f $fmt $file_name -q 2>/dev/null & process_id=$!
+    # delay for process run
+    sleep 1
+    # check process status is correct
+    sof-process-state.sh $process_id
+    [[ $? -ne 0 ]] && dloge "$cmd error without process" && exit 1
```
Comment (xiulipan) [inline test-case/check-suspend-resume-with-audio.sh]: `[[ $? -eq 1 ]]`?

File: test-case/check-suspend-resume-with-audio.sh [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+    $(dirname ${BASH_SOURCE[0]})/check-suspend-resume.sh $(echo $opt) 
+    ret=$?
+    [[ $ret -ne 0 ]] && dloge "suspend resume failed" && exit $ret
+    # check process status is correct
+    sof-process-state.sh $process_id
+    [[ $? -ne 0 ]] && dloge "process status is abnormal" && exit 1
+    kill -9 $process_id
+    sof-kernel-log-check.sh 0
+    [[ $? -ne 0 ]] && dloge "Catch dmesg error" && exit 1
```
Comment (xiulipan) [inline test-case/check-suspend-resume-with-audio.sh]: Do we also need to change this to`[[ $? -eq 1 ]]`

**Comment by Bin-QA** [comment]
> 2: process not exist
> 2: parameter error
> commit message got a replicated ‘2’

yes, in the script it exit with 2
from man bash
```
All builtins return an exit status of 2  to  indicate  incorrect usage, 
generally invalid options or missing arguments.
```

**Comment by Bin-QA** [comment]
Update for rebase

**Pull request merged**